src/pages/index.js

```diff
@@ -5,77 +5,210 @@ import PopupWithImage from "../components/PopupWithImage.js";
+    console.error(err);
+  });
+
 //FUNCTIONS----------------------
 
+function setButtonText(button, text) {
+  button.textContent = text;
+}
+
 function createCard(cardData) {
-  const card = new Card(cardData, "#element-template", () =>
-    handleImageClick(cardData)
+  const card = new Card(
+    cardData,
+    "#element-template",
+    () => handleImageClick(cardData),
+    handleDeleteButton,
+    handleCardLike
   );
   return card.getView();
 }
 
-function renderCard(cardData) {
-  const card = createCard(cardData);
-  cardsSection.addItem(card);
-}
+// function renderCard(cardData) {
+//   const card = createCard(cardData);
+//   section.addItem(card);
+// }
+
+//Image Popup Render
+
+const imagePopup = new PopupWithImage("#modal-preview");
+imagePopup.setEventListeners();
 
 function handleImageClick(cardData) {
   imagePopup.open(cardData.name, cardData.link);
 }
 
-cardsSection.renderItems();
+function handleCardLike(card) {
+  api
+    .likeCard(card._id, card.isLiked())
+    .then(() => {
+      card.handleLikeIcon();
+    })
+    .catch(console.error);
+}
 
-//User Info Render
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.setLoading(true);
+  return api
+    .updateProfileInfo(name, about)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .finally(() => {});
+}
 
-const userInfo = new UserInfo(".profile__name", ".profile__description");
+function handleAddNewCardSubmit(cardData) {
+  // renderCard(cardData, cardsWrap);
+  api
+    .createCard(cardData)
+    .then((res) => {
+      console.log(res);
+      // const card = createCard(cardData);
+      createCard(res);
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+//Confirm Delete Popup Render
 
-function handleProfileEditSubmit(userData) {
-  // const userData = editProfilePopup.getInputValues();
-  userInfo.setUserInfo(userData);
-  editProfilePopup.close();
+const confirmDelete = new PopupWithConfirmation("#confirm-modal");
+confirmDelete.setEventListeners();
+
+function handleDeleteButton(card) {
+  console.log("Delete button clicked");
+  confirmDelete.open();
+  console.log(card._id);
+  confirmDelete.setSubmitAction(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmDelete.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  });
+  console.log(card);
 }
 
-function handleAddNewCardSubmit(cardData) {
-  renderCard(cardData, cardsWrap);
-  newCardPopup.close();
+//Handle Profile Pic Submit
+function handleAvatarFormSubmit(inputValues) {
+  setButtonText(avatarModalButton, "Saving...");
+  console.log(inputValues);
+  api
+    .updateAvatar(inputValues.link)
+    .then((res) => {
+      userInfo.setUserAvatar(inputValues.link);
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err).finally(() => {
```

Comment:
Needs correction: `console.log` doesn't return promise. You should add `finally` block after `catch` in the api promise.